Armadillo/src/main/java/com/scribd/armadillo/ArmadilloPlayerChoreographer.kt

```diff
@@ -189,22 +193,37 @@ internal class ArmadilloPlayerChoreographer : ArmadilloPlayer {
 
     @Inject
     internal lateinit var context: Context
+
     @Inject
     internal lateinit var downloadEngine: DownloadEngine
+
     @Inject
     internal lateinit var cacheManager: CacheManager
+
     @Inject
     internal lateinit var stateProvider: StateStore.Provider
+
     @Inject
     internal lateinit var stateModifier: StateStore.Modifier
+
     @Inject
     internal lateinit var actionTransmitter: PlaybackActionTransmitter
+
     @Inject
     internal lateinit var mediaContentSharer: ArmadilloMediaBrowse.ContentSharer
 
-    private companion object {
-        const val TAG = "ArmadilloChoreographer"
+    companion object {
+        private const val TAG = "ArmadilloChoreographer"
         private val observerPollIntervalMillis = 500.milliseconds
+
+        @VisibleForTesting(otherwise = PRIVATE)
+        val handlerThread: HandlerThread by lazy {
+            HandlerThread("ArmadilloChoreographer")
+                .also { it.start() }
+        }
+
+        @VisibleForTesting(otherwise = PRIVATE)
+        val handler: Handler by lazy { Handler(handlerThread.looper) }
```

Comment:
You probably want to enforce what thread this happens on - if it gets lazily initiated from a random thread, that's the thread everything else is going to run on.

Reply:
it's all using the same HandlerThread